src/Language/GraphQL/Draft/Syntax.hs

```diff
@@ -315,9 +316,13 @@ type TypeCondition = NamedType
 --   | VObject !ObjectValue
 --   deriving (Ord, Show, Eq, Lift)
 
+-- Orphane instance allowing us to use Scientific values in Template Haskell
+-- Oxford brackets [| ... |]
+instance Lift S.Scientific where
+  lift sc = return (LitE (RationalL (toRational sc)))
```

Comment:
Orphan instances in libraries are usually a no-no—if some other dependency defines the same orphan instance, everything fails to compile, and there’s nothing you can do to fix it. Can we do the lifting manually where it’s needed?

Reply:
In fact, I don't think this particular instance is ever used: GHC asked me to provide this instance because of the derived `Lift` instance for `ValueConst` and `Value`.

Reply:
I think some code in `graphql-engine` depends on the existence of those derived `Lift` instances, so it would be nice to keep them! But we can write them by hand instead of deriving them so that we can avoid the `Lift Scientific` orphan.

Reply:
(Another option would be to submit a PR to the `scientific` package to provide a `Lift` instance, which might be a good idea to do anyway, but it would be a little more involved.)

Reply:
In the `scientific` package one can derive the `Lift` instance. So that's certainly the right way to do it.

Do we have enough patience to wait for that to subsequently get released? I suppose this would also depend on the `scientific` authors wanting to depend on `template-haskell` and `+XDeriveLift`. Otherwise I'll write the `Lift ValueConst` and `Lift Value` instances by hand.

Reply:
> I suppose this would also depend on the `scientific` authors wanting to depend on `template-haskell` and `+XDeriveLift`.

Depending on `template-haskell` is essentially “free,” since `template-haskell` is a wired-in package, so it seems unlikely the maintainers would care. (And everyone uses GHC these days, so depending on `DeriveLift` is probably not a big deal.)

> Do we have enough patience to wait for that to subsequently get released?

We could depend on a fork of `scientific` until the changes get upstreamed. But yes, it would be more involved, so writing the instance by hand is quite alright, too.